model/crud/crud.go

```diff
@@ -0,0 +1,29 @@
+// Package crud provides a crud interface
+package crud
+
+type CRUD interface {
+	// Read values
+	Read(...ReadOption) ([]Record, error)
+	// Write a value
+	Write(id string, v interface{}) error
+	// Update a record
+	Update(id string, v interface{}) error
+	// Delete a record
+	Delete(id string) error
+}
+
+type Record interface {
+	// Value
+	Value() []byte
+	// Scan the value into an interface
+	Scan(v interface{}) error
+}
+
+type ReadOption struct {
+	// Id to read
+	Id string
+	// Limit number of returns records
+	Limit int
+	// Offset
+	Offset int
+}
```

Comment:
e.g., ListOption should have offset and limit too

Reply:
There is no list. CRUD doesn't have anything called list. You use Read and it returns rows.